app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java

```diff
@@ -179,12 +179,17 @@ public void onNothingSelected(AdapterView<?> adapterView) {
                     // If user has chosen a default language from settings activity savedLanguageValue is not null
                     spinnerDescriptionLanguages.setSelection(languagesAdapter.getIndexOfLanguageCode(savedLanguageValue));
                 } else {
-                    if (position == 0) {
-                        int defaultLocaleIndex = languagesAdapter
-                                .getIndexOfUserDefaultLocale(spinnerDescriptionLanguages.getContext());
-                        spinnerDescriptionLanguages.setSelection(defaultLocaleIndex, true);
+                    if(uploadMediaDetails.get(position).getLanguageCode() != null){
```

Comment:
Would you mind adding a comment line before this block, to explain in what circumstances it is reached, and what the code does? Thanks!

Reply:
I have added it,  code is just providing the same previous language code to the spinner for current upload, when user presses the copy previous caption & description button .